api/src/controllers/elastic-search.ts

```diff
@@ -0,0 +1,130 @@
+import { Client } from "@elastic/elasticsearch";
+import * as prismic from "@prismicio/client";
+import {
+  Article,
+  ArticlePrismicDocument,
+  Clients,
+  ContentType,
+} from "../types";
+import {
+  CountResponse,
+  ErrorCause,
+} from "@elastic/elasticsearch/lib/api/types";
+import { Config, getConfig } from "../../config";
+import { articlesContentTypes, graphQueryArticles } from "../helpers/articles";
+import { transformArticle } from "../transformers/article";
+import { HttpError } from "./error";
+import { PaginationQueryParameters } from "./pagination";
+import { RequestHandler } from "express";
+
+const indexName = getConfig().contentsIndex;
+
+type QueryParams = {
+  query?: string;
+} & PaginationQueryParameters;
+
+type Handler = RequestHandler<
+  never,
+  { count: CountResponse },
+  never,
+  QueryParams
+>;
+
+// TODO remove?
+const addIndex = async (elasticClient: Client, indexName: string) => {
+  const exists = await elasticClient.indices.exists({ index: indexName });
+
+  if (!exists) {
+    await elasticClient.indices.create({ index: indexName });
+    console.log(indexName, "was created");
+  } else {
+    console.log("Index", indexName, "already exists");
+  }
+};
+
+const useBulkHelper = async (elasticClient: Client, docs: Article[]) => {
+  const operations = docs.flatMap((doc) => {
+    return [
+      { index: { _index: indexName, _id: doc.id } },
+      {
+        display: doc as Article,
+        query: { id: doc.id },
+      },
+    ];
+  });
+
+  const bulkResponse = await elasticClient.bulk({ refresh: true, operations });
```

Comment:
I would strongly suggest that you use the [bulk helper](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/client-helpers.html) instead of the raw API method - there are things it does under the hood around error checking and proper streaming that will be a pain to do ourselves.